lua/entities/gmod_tardis/modules/sh_destination.lua

```diff
@@ -92,22 +100,83 @@ TARDIS:AddKeyBind("destination-demat",{
     clientonly=true,
     exterior=true
 })
+TARDIS:AddKeyBind("destination-jump",{
+    name="Jump",
+    section="Destination",
+    func=function(self,down,ply)
+        if ply:GetTardisData("destination") then
+            if down then
+                self:SetData("destination-trace",true)
+            elseif self:GetData("destination-trace") then
+                local prop = self:GetData("destinationprop")
+                local pos,ang = self:GetDestinationPropTrace(ply,ply:EyeAngles())
+
+                prop:SetPos(pos)
+                prop:SetAngles(ang)
+
+                self:SetData("destination-trace",false)
+            end
+        end
+    end,
+    key=MOUSE_RIGHT,
+    clientonly=true,
+    exterior=true
+})
 TARDIS:AddKeyBind("destination-snaptofloor",{
     name="SnapToFloor",
     section="Destination",
     func=function(self,down,ply)
-        if ply:GetTardisData("destination") then
+        if down and ply:GetTardisData("destination") then
             local prop = self:GetData("destinationprop")
             if IsValid(prop) then
-                local pos = util.QuickTrace(prop:GetPos()+Vector(0,0,50), Vector(0,0,-1)*99999999).HitPos
+                local prevpos = self:GetData("destination_snaptofloor_lastpos")
+                local should_glue = (prevpos == prop:GetPos())
```

Comment:
In playtesting I found there were several places like here on `freespace_revolution` where this seems to work very inconsistently
![image](https://github.com/MattJeanes/TARDIS/assets/2363642/796af666-c6b5-4336-b9fd-20385e8b6a78)

Reply:
c206c2b812d5ccd92f8922c939c71fbdf3bc505d
45b84a5536488bd6a5decf61fc479fc782cf91bb
I did the best I could

Plus, allowed SHIFT+R for another approach (same as right click), it's gonna be a rare use-case but could come in handy so no reason not to have it